PSReadLine/ReadLine.cs

```diff
@@ -196,12 +200,17 @@ internal static PSKeyInfo ReadKey()
                     //   - a key is pressed
                     //   - the console is exiting
                     //   - 300ms - to process events if we're idle
-                    //   - processing of events is requested externally
                     //   - ReadLine cancellation is requested externally
                     handleId = WaitHandle.WaitAny(_singleton._requestKeyWaitHandles, 300);
                     if (handleId != WaitHandle.WaitTimeout && handleId != EventProcessingRequested)
                         break;
 
+                    if (_handleIdleOverride is not null)
```

Comment:
By removing `_forceEventWaitHandle`, how do you break out `WaitHandle.WaitAny` immediately when you need to? Are you planning to use the cancellation token for that purpose?